pkg/log/httpbuffer/httpbuffer.go

```diff
@@ -0,0 +1,101 @@
+package httpbuffer
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sync"
+
+	"net/http"
+
+	"github.com/go-kit/kit/log"
+)
+
+type httpBuffer struct {
+	buf                 bytes.Buffer
+	bufMutex, sendMutex sync.Mutex
+	maxSize, sendSize   int
+	endpoint            string
+	logger              log.Logger
+}
+
+type httpBufferOption func(*httpBuffer)
+
+func WithMaxSize(maxSize int) httpBufferOption {
+	return func(hb *httpBuffer) {
+		hb.maxSize = maxSize
+	}
+}
+
+func WithSendSize(sendSize int) httpBufferOption {
+	return func(hb *httpBuffer) {
+		hb.sendSize = sendSize
+	}
+}
+
+func WithLogger(logger log.Logger) httpBufferOption {
+	return func(hb *httpBuffer) {
+		hb.logger = logger
+	}
+}
+
+func New(endpoint string, opts ...httpBufferOption) *httpBuffer {
```

Comment:
instead of accepting a string endpoint, probably accept an http client or perhaps a custom interface that will handle auth and all that